okhttp/src/main/java/okhttp3/internal/connection/RealConnection.kt

```diff
@@ -690,7 +690,7 @@ class RealConnection(
       if (e is StreamResetException) {
         when (e.errorCode) {
           ErrorCode.REFUSED_STREAM -> {
-            // Retry REFUSED_STREAM errors once on the same connection.
+            // Stop using this connection after one REFUSED_STREAM error.
             refusedStreamCount++
             if (refusedStreamCount > 1) {
```

Comment:
The ExchangeFinder counts now mirror similar counts on RealConnection. We need to track per-connection failures and per-exchange failures separately!